ClangPowerTools/ClangPowerTools/ClangPowerToolsPackage.cs

```diff
@@ -261,12 +266,33 @@ public int OnAfterCloseSolution(object aPUnkReserved)
       return VSConstants.S_OK;
     }
 
-
     #endregion
 
 
     #region Private Methods
 
+    private bool SolutionDoesNotContainCppProject()
```

Comment:
I think it's better to use another class for this solution check (Separation of concerns principle & S from SOLID).

Reply:
I added a SolutionManager class with a method that checks if there are cpp projects in solution.